rs/ic_os/hostos_tool/src/main.rs

```diff
@@ -73,8 +74,14 @@ pub fn main() -> Result<()> {
                 }
             };
 
+            let mgmt_mac: Option<&str> = match &deployment {
```

Comment:
Maybe, could we do the parsing here so that we have the typed `FormattedMacAddress` earlier?

Reply:
I started on this but then the changes started to really balloon, so I made a separate branch that I can pick up after this PR merges: https://github.com/dfinity/ic/pull/1397/files (see latest commit)

WDYT?